feature/platform/ate_tests/telemetry_inventory_test/telemetry_inventory_test.go

```diff
@@ -440,119 +412,97 @@ func ValidateComponentState(t *testing.T, dut *ondatra.DUTDevice, cards []string
 				t.Errorf("component.SerialNo().Get(t) for %q): got empty string, want non-empty string", card)
 			}
 		}
-
 		if p.mfgNameValidation {
 			mfgName := gnmi.Get(t, dut, component.MfgName().State())
-			t.Logf("Hardware card %s mfgName: %s", card, mfgName)
 			if mfgName == "" {
 				t.Errorf("Get mfgName for %q): got empty string, want non-empty string", card)
 			}
 		}
-
 		if p.mfgDateValidation {
 			mfgDate := gnmi.Get(t, dut, component.MfgDate().State())
-			t.Logf("Hardware card %s mfgDate: %s", card, mfgDate)
 			if mfgDate == "" {
 				t.Errorf("component.MfgName().Get(t) for %q): got empty string, want non-empty string", card)
 			}
 		}
-
 		if p.swVerValidation {
 			// Only a subset of cards are expected to report Software Version.
 			sw, present := gnmi.Lookup(t, dut, component.SoftwareVersion().State()).Val()
 			if present {
-				t.Logf("Hardware card %s SoftwareVersion: %s", card, sw)
 				if sw == "" {
 					t.Errorf("component.SoftwareVersion().Get(t) for %q): got empty string, want non-empty string", card)
 				}
 			} else {
 				t.Errorf("component.SoftwareVersion().Lookup(t) for %q): got no value.", card)
 			}
 		}
-
 		if p.hwVerValidation {
 			hardwareVersion := gnmi.Get(t, dut, component.HardwareVersion().State())
-			t.Logf("Hardware card %s hardwareVersion: %s", card, hardwareVersion)
 			if hardwareVersion == "" {
 				t.Errorf("component.HardwareVersion().Get(t) for %q): got empty string, want non-empty string", card)
 			}
 		}
-
 		if p.fwVerValidation {
 			firmwareVersion := gnmi.Get(t, dut, component.FirmwareVersion().State())
-			t.Logf("Hardware card %s FirmwareVersion: %s", card, firmwareVersion)
 			if firmwareVersion == "" {
 				t.Errorf("component.FirmwareVersion().Get(t) for %q): got empty string, want non-empty string", card)
 			}
 		}
-
 		if p.rrValidation {
 			redundantRole := gnmi.Get(t, dut, component.RedundantRole().State()).String()
-			t.Logf("Hardware card %s RedundantRole: %s", card, redundantRole)
 			if redundantRole != "PRIMARY" && redundantRole != "SECONDARY" {
 				t.Errorf("component.RedundantRole().Get(t) for %q): got %v, want %v", card, redundantRole, p.operStatus)
 			}
 		}
-
 		if p.operStatus != "" {
 			operStatus := gnmi.Get(t, dut, component.OperStatus().State()).String()
-			t.Logf("Hardware card %s OperStatus: %s", card, operStatus)
 			if operStatus != activeStatus {
 				t.Errorf("component.OperStatus().Get(t) for %q): got %v, want %v", card, operStatus, p.operStatus)
 			}
 		}
-
 		if p.parent != "" {
-			parent := gnmi.Get(t, dut, component.Parent().State())
-			t.Logf("Hardware card %s parent: %s", card, parent)
-			if parent != p.parent {
-				t.Errorf("component.Parent().Get(t) for %q): got %v, want %v", card, parent, p.parent)
+			parent := gnmi.Lookup(t, dut, gnmi.OC().Component(card).Parent().State())
+			val, present := parent.Val()
+			if !present {
+				t.Errorf("Chassis component NOT found in the hierarchy tree of component %s", card)
```

Comment:
can use %q for the card, as you do in the error below

Reply:
> can use %q for the card, as you do in the error below

Done